app/src/main/java/net/tensory/rxjavatalk/providers/BattleProvider.java

```diff
@@ -4,16 +4,24 @@
 import net.tensory.rxjavatalk.data.DragonManager;
 import net.tensory.rxjavatalk.models.Battle;
 
+import java.util.concurrent.TimeUnit;
+
 import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
 
 public class BattleProvider {
 
-    private BattleFrontFeed northernFrontFeed;
-    private BattleFrontFeed southernFrontFeed;
+    private PublishSubject<Battle> battlesSubject = PublishSubject.create();
 
     public BattleProvider(DragonManager dragonManager) {
-        northernFrontFeed = new BattleFrontFeed(BattleFrontFeed.Front.NORTHERN, dragonManager);
-        southernFrontFeed = new BattleFrontFeed(BattleFrontFeed.Front.SOUTHERN, dragonManager);
+        BattleFrontFeed northernFrontFeed = new BattleFrontFeed(BattleFrontFeed.Front.NORTHERN, dragonManager);
+        BattleFrontFeed southernFrontFeed = new BattleFrontFeed(BattleFrontFeed.Front.SOUTHERN, dragonManager);
+        Observable.merge(
+                northernFrontFeed.observeBattles(),
+                southernFrontFeed.observeBattles()
+                        .delay(2, TimeUnit.SECONDS)
```

Comment:
I added this because, with both streams firing at a random interval, I would expect to see only two houses update at once -- but if the firings are nearly simultaneous, the highlighting may make it look like 2, 3, or 4 houses are all lighting up.

It's easier to see that the Observable<Battle> bug was fixed if the firing is staggered and regular.

However, this method for staggering the streams means that they interleave evenly: north, south, north, south... So I'm going to sleep on a random firing scheme that adds enough delay in between events so that the updates look right.

Reply:
My only concern with this method, is that it's our merge example and I'm a little worried about not having it pristine.  Maybe we can change the delay's on the FrontFeed, so that they are properly staggered